.github/workflows/echidna.yaml

```
@@ -0,0 +1,58 @@
+name: "Echidna"
+
+env:
+  FOUNDRY_PROFILE: "ci"
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - "main"
+      - "feature/**"
+      - "feat/**"
+      - "develop"
+  push:
+    branches:
+      - "main"
+      - "develop"
+
+jobs:
+  get_echidna_test_names:
+    name: Get Echidna test names
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Generate matrix with all test names from the test/echidna directory
+        id: set-matrix
+        run: |
+          echo "::set-output name=matrix::$(ls test/echidna | sed 's/.sol//' | jq -R -s -c 'split("\n")[:-1]')"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  run_echidna_tests:
+    needs: get_echidna_test_names
+    name: Run echidna tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        contract: ${{ fromJson(needs.get_echidna_test_names.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: "nightly-d369d2486f85576eec4ca41d277391dfdae21ba7"
```

while we're at it, should we update to latest foundry release:
[nightly-fd87629fbc4ae2e0fa00ccf42b4a9ebe1b521d55](https://github.com/foundry-rs/foundry/releases/tag/nightly-fd87629fbc4ae2e0fa00ccf42b4a9ebe1b521d55)

```suggestion
          version: "nightly-fd87629fbc4ae2e0fa00ccf42b4a9ebe1b521d55"
```